### PR TITLE
Use linuxdeployqt

### PR DESCRIPTION
**Pull request by patrickelectric**
Use linuxdeployQt to create AppImage
List:

* [x] Create .appimage
* [x] Gst working
* [x] Test drone connection and usage

Download appimage here: https://file.io/d5eiAZ

### CODE REVIEW

**Comment by dagar** [comment]
Looks good. Can we do this on Jenkins instead of Travis-ci?

**Comment by patrickelectric** [comment]
@dagar I can do that after finishing this PR.

**Comment by patrickelectric** [comment]
Im using transfer to test the appimage. This is still WIP.

File: .travis.yml [hunk 1]
```diff
@@ -189,11 +189,12 @@ after_success:
...
     fi
 
-before_deploy:
+  #before_deploy:
   # create linux appimage
   - cd ${TRAVIS_BUILD_DIR}
   - if [[ "${SPEC}" = "linux-g++-64" && "${CONFIG}" = "installer" ]]; then
-        ./deploy/create_linux_appimage.sh ${TRAVIS_BUILD_DIR} ${SHADOW_BUILD_DIR}/release ${SHADOW_BUILD_DIR}/release/package;
+        ./deploy/create_linux_appimage_with_linuxdeployqt.sh ${SHADOW_BUILD_DIR};
+        curl --upload-file ./QGroundControl-x86_64.AppImage https://transfer.sh/QGroundControl-x86_64.AppImage;
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply .travis.yml]: As I said here:https://github.com/mavlink/qgroundcontrol/pull/6946#issuecomment-438308161
This is only for test.

**Comment by patrickelectric** [comment]
@dagar @DonLakeFlyer 	@dogmaphobic 	, it appears to be working here.
The most important things to test is: video connection and parameter download.

**Comment by dagar** [comment]
Sorry this slipped through. Still interested?

**Comment by patrickelectric** [comment]
Hi, sorry we are a bit busy right now, the only thing that is missing is to
include gst dynamic libraries. but everything else appears to work

**Comment by patrickelectric** [comment]
@tcanabrava are you interested to finish this ? Otherwise I can finish it in the next 1~2 months. 
There is also [linuxdeploy](https://github.com/linuxdeploy/linuxdeploy) if you want to try.

**Comment by DonLakeFlyer** [comment]
So I took a look at linuexdeployqt. The problem with it is:
* I would want this to build an appimage from the build system not travis
* But linuxdeployqt is crazily anal about requiring a base version of Ubuntu to build from. With no way to work around it. And if you don't have that it simply doesn't work. So for example if you have installed the PX4 dev tools it updates some stuff beyond the base Ubuntu 18.04 image. Which in turns means linuxdeployqt simple refuses to run. So although it would work fine with the vanilla travis image of Xenial. It won't work on local machines.

**Comment by patrickelectric** [comment]
@DonLakeFlyer I agree with you, this changed from the last year and a new toll based in one of linuxdeployqt main devs started: https://github.com/linuxdeploy/linuxdeploy
And I believe that this tool will probably accomplish what linuxdeployqt fails to do.

**Comment by DonLakeFlyer** [comment]
Closing since we can't use linuxdeplyqt

**Pull request closed without merging**